### ISSUE TITLE
[Bug] `getContentCards` doesn't trigger a refresh on iOS

### ISSUE THREAD

**Issue by Minishlink** [braze-inc/braze-react-native-sdk#195]
### Which Platforms?

iOS

### Which React Native Version?

0.70

### Which @braze/react-native-sdk SDK version?

2.0.2

### Repro Rate

100%

### Steps To Reproduce

Flow A:
  1. `import Braze from "@braze/react-native-sdk"`.
  2. `Braze.getContentCards().then(console.log)`
  2. Run the app.
 
Flow B:
  1. `import Braze from "@braze/react-native-sdk"`.
  2. `Braze.getContentCards()` (with the workaround below)
  3. `Braze.logContentCardDismissed(cardId)`
  4. Call `Braze.getContentCards()`

### Expected Behavior

Example A : It should return fresh content cards like on Android.
Example B : The dismissed card is dismissed like on Android.

### Actual Incorrect Behavior

Flow A: It returns nothing. 
You have to use a workaround to do what the Android SDK is doing

```ts
await new Promise<void>(resolve => {
    const listener = Braze.addListener(Braze.Events.CONTENT_CARDS_UPDATED, () => {
       listener.remove();
          resolve();
     });
     Braze.requestContentCardsRefresh();
});
await Braze.getContentCards();
```

Flow B: The dismissed card isn't dismissed
Workaround is to call Braze.requestContentCardsRefresh() after logDismiss (but subject to race condition)

**Comment by hokstuff** [comment]
Hi @Minishlink,

We have an [example in our sample app](https://github.com/braze-inc/braze-react-native-sdk/blob/250315db070c9795acd05c28e57efe15d4990466/BrazeProject/BrazeProject.tsx#L430) using the `getContentCards()` API. You can work with that sample app with your own API key and endpoint to try to understand the sample's current behavior.

Can you contact support@braze.com with more information about:
- The Content Cards you are expecting to be returned in Flow A
  - We need to determine if the data is actually coming down to the device but the `getContentCards()` is not returning the correct info, if `getContentCards()` is getting cached data, or something else
- Whether or not you are flushing the data to the server after an interaction (e.g. dismiss card)

Thanks!

**Comment by Minishlink** [comment]
You can see clearly the difference of behaviour between Android and iOS here:

* android
  * https://github.com/braze-inc/braze-react-native-sdk/blob/master/android/src/main/java/com/braze/reactbridge/BrazeReactBridge.kt#L377
  * it correctly calls a refresh
* ios
  * https://github.com/braze-inc/braze-react-native-sdk/blob/master/iOS/BrazeReactBridge/BrazeReactBridge/BrazeReactBridge.m#L546
  * it doesn't call a refresh

**Comment by patelbansari** [comment]
Facing same issue on iOS with latest `"@braze/react-native-sdk": "^2.1.0".` The only way I see the new content card is if I click the content card's push notification from the notification center.

**Comment by hokstuff** [comment]
Hi @Minishlink,

Thanks for linking the code snippets. We have filed this issue internally and will slate the change to be done in a future release.

**Comment by hokstuff** [comment]
Hi @Minishlink,

We have released [version 3.0.0](https://github.com/braze-inc/braze-react-native-sdk/releases/tag/3.0.0) which addresses this issue that was introduced in version 2.0.0. The behavior on iOS should now match Android's behavior when using custom UI. Let us know if you run into issues - thanks!

**Comment by WojciechRupik** [comment]
Guys, i am struggling with same issue. I got cards data on android but i can't get it on ios. Currently i am working on simulator but i shouldn't be a problem. 

I am using this version: 
`"@braze/react-native-sdk": "12.1.0"`

[ios code](https://github.com/braze-inc/braze-react-native-sdk/blob/3.0.0/iOS/BrazeReactBridge/BrazeReactBridge/BrazeReactBridge.m#L516)

[android code](https://github.com/braze-inc/braze-react-native-sdk/blob/3.0.0/android/src/main/java/com/braze/reactbridge/BrazeReactBridge.kt#L377)
